### ISSUE TITLE
Trivy operator is not scanning the image but not generating the vulnerbility report in the gke cluster

### ISSUE THREAD

**Issue by saurabh21316** [aquasecurity/trivy-operator#1668]
Hello @chen-keinan 

Hope you doing well. It's been long time talked to you. I see a new issue with trivy operator where it's not generating report however I see it's running job/pod in the gke cluster. For example., Running locally trivy command works

docker run --rm -v /Users/trivy/service_account:/temp -e GOOGLE_APPLICATION_CREDENTIALS=/temp/docker-image-17ed6db8acfe.json aquasec/trivy:0.47.0 image --format json -o /temp/output.json docker.elastic.co/elasticsearch/elasticsearch:8.11.1

So I mean I have three different elastic search clusters running in same namespace on gke cluster.
with version 8.9, 8.10 and 8.11 and I see vulnerbility report generated for 8.9 and 8.10 but missing 8.11

I have another example where this image docker.io/apachepulsar/pulsar-all:2.11.1  is being used by multiple statefulset in the namespace but not generating even a single report.

Something is broken it seems like. If you can look into this that would be really helpful.

Thanks!

**Comment by chen-keinan** [comment]
@saurabh21316 thanks for the feedback , I'll take a look and update you

**Comment by chen-keinan** [comment]
@saurabh21316 are you using gcr service account setting as in [docs](https://aquasecurity.github.io/trivy-operator/latest/tutorials/private-registries/#fifth-option-configure-gcr-service-account-json)

**Comment by saurabh21316** [comment]
@chen-keinan I can confirm that it's not an issue with gcr account because my private images are getting scanned. I feel like it's more of issue about missing report after pod/job finish scanning.

**Comment by saurabh21316** [comment]
Also  see this kind of error and it sutcks in crashloop 

2023-12-04 21:59:15.652Z {"level":"error","ts":"2023-12-04T21:59:15Z","msg":"Reconciler error","controller":"networkpolicy","controllerGroup":"networking.k8s.io","controllerKind":"NetworkPolicy","NetworkPolicy":{"name":"dev-exporter-cradlepoint-standalone","namespace":"colony-testing"},"namespace":"colony-testing","name":"dev-exporter-cradlepoint-standalone","reconcileID":"4ddc40f6-463e-434b-a4be-c562f01906d4","error":"evaluating resource: context canceled","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}
2023-12-04 21:59:15.903Z {"level":"error","ts":"2023-12-04T21:59:15Z","msg":"Reconciler error","controller":"networkpolicy","controllerGroup":"networking.k8s.io","controllerKind":"NetworkPolicy","NetworkPolicy":{"name":"access-to-pulsar","namespace":"colony-logging"},"namespace":"colony-logging","name":"access-to-pulsar","reconcileID":"a291fa74-4778-48ed-b43d-5321b3360c76","error":"evaluating resource: context canceled","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}

**Comment by chen-keinan** [comment]
@saurabh21316 I see this error related to `network policy` config evaluation

**Comment by chen-keinan** [comment]
> networkpolicy

sure I'll do the effort

**Comment by saurabh21316** [comment]
Thanks!! @chen-keinan

How about missing reports?

**Comment by saurabh21316** [comment]
Hello @chen-keinan Just checking if you get a chance to look into missing vulnerable reports. I am currently blocked on it.

**Comment by chen-keinan** [comment]
@saurabh21316 the errors I see in the logs are related to `networkpolicy` could be these issue also prevent from reports being generated

**Comment by chen-keinan** [comment]
@saurabh21316 can you try extend [scan-job timeout](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/values.yaml#L57), [trivy timeout](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/values.yaml#L365)  and also use fast scan by setting this [flag](https://github.com/aquasecurity/trivy-operator/blob/main/deploy/helm/values.yaml#L350) as well

**Comment by saurabh21316** [comment]
@chen-keinan  I extended time to 10mins for both timeout and regarding default flag slow:true,  I changed it to false.

**Comment by saurabh21316** [comment]
Let's see if it helps

**Comment by saurabh21316** [comment]
@chen-keinan just an update it seems like it didn't help... I tried setting slow: true and false but no luck.

**Comment by chen-keinan** [comment]
Thanks for the update

**Comment by saurabh21316** [comment]
@chen-keinan what kind of error are these?

2023-12-28 20:27:48.814Z {"level":"error","ts":"2023-12-28T20:27:48Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-685fdd6458","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-685fdd6458","reconcileID":"644bbbbd-5a9d-49a8-90f6-3ac6a56969aa","error":"unexpected EOF","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}
2023-12-28 20:44:28.912Z {"level":"error","ts":"2023-12-28T20:44:28Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-685fdd6458","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-685fdd6458","reconcileID":"8aeb47a5-f04f-44bd-a7a3-384d084e05ff","error":"unexpected EOF","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

**Comment by chen-keinan** [comment]
Can you please share your config maps and deployments envvar ?

**Comment by saurabh21316** [comment]
config map
[trivy-configmap.zip](https://github.com/aquasecurity/trivy-operator/files/13789732/trivy-configmap.zip)

helm
[trivy-operator-helm.yaml.zip](https://github.com/aquasecurity/trivy-operator/files/13789782/trivy-operator-helm.yaml.zip)

@chen-keinan

**Comment by saurabh21316** [comment]
Another type error:

2023-12-29 00:15:46.327Z {"level":"error","ts":"2023-12-29T00:15:46Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-6dd867fcd9","container":"pulsar-toolset","status.reason":"Error","status.message":"2023-12-29T00:15:41.951Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:dec9a325ebd3178f53a8bba1c481ed3fd9b9c5a080c7d541f7b3ceae9ab1b52d): walk error: failed to process the file: failed to analyze file: failed to analyze var/lib/dpkg/info/libssl1.1:amd64.symbols: unable to open var/lib/dpkg/info/libssl1.1:amd64.symbols: failed to open: failed to copy: stream error: stream ID 5; INTERNAL_ERROR; received from peer\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:290\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:81\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}
2023-12-29 00:23:01.753Z {"level":"error","ts":"2023-12-29T00:23:01Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-7696bcfb84","container":"pulsar-recovery","status.reason":"Error","status.message":"2023-12-29T00:15:01.844Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:dec9a325ebd3178f53a8bba1c481ed3fd9b9c5a080c7d541f7b3ceae9ab1b52d): walk error: failed to extract the archive: stream error: stream ID 5; INTERNAL_ERROR; received from peer\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:290\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:81\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

**Comment by saurabh21316** [comment]
2023-12-29 01:55:36.620Z {"level":"error","ts":"2023-12-29T01:55:36Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-5f75655f9","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-5f75655f9","reconcileID":"94db73c3-e7d4-41b4-9038-0d63e6d5c0da","error":"etcdserver: request is too large","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}
2023-12-29 01:58:47.421Z {"level":"error","ts":"2023-12-29T01:58:47Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-69d8468589","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-69d8468589","reconcileID":"1728ee57-7cc1-442e-8987-c364f7bc718d","error":"an empty namespace may not be set during creation","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

**Comment by chen-keinan** [comment]
> Different Kind of Error:
> 
> 2023-12-29 01:55:36.620Z {"level":"error","ts":"2023-12-29T01:55:36Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-5f75655f9","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-5f75655f9","reconcileID":"94db73c3-e7d4-41b4-9038-0d63e6d5c0da","error":"etcdserver: request is too large","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"} 2023-12-29 01:58:47.421Z {"level":"error","ts":"2023-12-29T01:58:47Z","msg":"Reconciler error","controller":"job","controllerGroup":"batch","controllerKind":"Job","Job":{"name":"scan-vulnerabilityreport-69d8468589","namespace":"trivy-system"},"namespace":"trivy-system","name":"scan-vulnerabilityreport-69d8468589","reconcileID":"1728ee57-7cc1-442e-8987-c364f7bc718d","error":"an empty namespace may not be set during creation","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

this is a cluster limitation  see open issue #441

**Comment by chen-keinan** [comment]
> Different Error Type:
> 
> 2023-12-29 00:15:46.327Z {"level":"error","ts":"2023-12-29T00:15:46Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-6dd867fcd9","container":"pulsar-toolset","status.reason":"Error","status.message":"2023-12-29T00:15:41.951Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:dec9a325ebd3178f53a8bba1c481ed3fd9b9c5a080c7d541f7b3ceae9ab1b52d): walk error: failed to process the file: failed to analyze file: failed to analyze var/lib/dpkg/info/libssl1.1:amd64.symbols: unable to open var/lib/dpkg/info/libssl1.1:amd64.symbols: failed to open: failed to copy: stream error: stream ID 5; INTERNAL_ERROR; received from peer\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:290\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:81\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"} 2023-12-29 00:23:01.753Z {"level":"error","ts":"2023-12-29T00:23:01Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-7696bcfb84","container":"pulsar-recovery","status.reason":"Error","status.message":"2023-12-29T00:15:01.844Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:dec9a325ebd3178f53a8bba1c481ed3fd9b9c5a080c7d541f7b3ceae9ab1b52d): walk error: failed to extract the archive: stream error: stream ID 5; INTERNAL_ERROR; received from peer\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:290\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:81\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

this is look like trivy issue, are you able to scan the image with trivy cli ?

**Comment by ODB686** [comment]
hi @chen-keinan , 
i faced the same issue when scanning one of our images (pretty big, 22gb) using both 0.23.0 and latest 0.27.0 trivy-operator versions in eks cluster. All other images were scanned well.

```
2025-06-17T11:19:11Z	ERROR	reconciler.scan job	Scan job container	{"job": "trivy-system/scan-vulnerabilityreport-7f4bc4b98f", "container": "test-service", "status.reason": "Error", "status.message": "2025-06-17T11:19:03Z\tFATAL\tFatal error\trun error: image scan error: scan error: scan failed: failed analysis: analyze error: pipeline error: failed to analyze layer (sha256:95b666b44afa2f736bb4e7ea9de46eaba22da70ad15ca126e37514794709d03e): walk error: failed to process the file: failed to analyze file: failed to analyze srv/test-service/app.jar: unable to open srv/test-service/app.jar: failed to open: failed to copy: stream error: stream ID 29; INTERNAL_ERROR; received from peer\n"}
github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).completedContainers
	/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:448
github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).SetupWithManager.(*ScanJobController).reconcileJobs.func1
	/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:85
sigs.k8s.io/controller-runtime/pkg/reconcile.TypedFunc[...].Reconcile
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/reconcile/reconcile.go:134
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Reconcile
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/internal/controller/controller.go:340
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/internal/controller/controller.go:300
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func2.1
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.21.0/pkg/internal/controller/controller.go:202
```

trivy image in cli mode scans this image quite fast and completely. Any suggestions?

Best regards,

**Comment by JanWelker** [comment]
I think I have the cause for the "image used by several workloads, no report" variant, and a remediation. Operator 0.34.0 / Trivy 0.74.0, self-hosted (not GKE, so the credentials angle earlier in this thread is not involved).

**It is `clusterSbomCacheEnabled`. The scans succeed and the vulnerability report is dropped.**

## The evidence that pointed at it

Of 84 workload containers with no `VulnerabilityReport` in my cluster:

| | count | state |
| --- | ---: | --- |
| **has an `SbomReport`** | **80** | scan ran and succeeded — only the vulnerability report was never written |
| has neither | 4 | report too large for the API server — the separate problem in #757 |

The SBOMs are timestamped with the scan. Those 80 are not failures.

## The path

When an image already has a `ClusterSbomReport`, the reconcile takes a reuse branch instead of a normal scan:

```go
reportsData = getGlobalSbomReports(ctx, r.SbomReadWriter, containerImages, log)
if len(reportsData) > 0 {
    err = r.reuseSbomReport(ctx, workloadObj, reportsData)   // writes the SbomReports here
}
r.SubmitScanJobChan <- ScanJobRequest{..., ClusterSbomReport: reportsData}
```

`reuseSbomReport` writes the namespaced `SbomReport`s up front, and the scan job that follows runs `trivy sbom` rather than `trivy image` to derive vulnerabilities from them. That second step produced nothing here, so the SBOM lands and the CVE data does not.

That is why it tracks images shared between workloads, which is the pattern this issue keeps describing. **The first workload to use an image does a real scan and gets its report; every workload after it finds a cached SBOM and takes this branch.** Both images affected in my cluster, `ceph/ceph:v20.2.4` and `cilium/cilium:v1.20.2`, were in the cluster SBOM cache.

## Remediation, and the measurement

Setting `operator.clusterSbomCacheEnabled: false`:

```
vulnerability reports   122 -> 205
containers missing       84 -> 4
rook-ceph missing        51 -> 0
```

The operator immediately began logging `Submitting a scan for the workload` for ReplicaSets it had not attempted once in the preceding 24 hours. The 4 that remain are the oversized-report case above, which is unrelated.

The cost is that a repeated image is decomposed once per workload instead of once per cluster — slower and redundant. Which seems much the cheaper mistake, since the cache was saving that work by discarding its result.

## Two things this is *not*, both of which I checked

**Not contention on the scan job's shared `/tmp/trivy/.cache`.** It is the obvious theory — every scan container mounts the same emptyDir and gets the same `--cache-dir` — and it is wrong. Trivy walks layers as a stream (`LayerTar.Walk(layer io.Reader, ...)`), so a shared cache directory cannot truncate a layer read. I ran six concurrent scans of the same 1.3 GB image against a shared cache deliberately: all six passed.

**Not the `unexpected EOF` scan failures.** Those are real and do appear in my logs, but they account for **5** containers out of 84, not 80. Worth separating, because they look like the same symptom from outside.

## A smaller observation

One reason this took a while: with `concurrentScanJobsLimit: 10` and `scanJobTTL: 10m`, a *completed* job still occupies a slot until the TTL reaps it, so a backlog drains at ten workloads per ten minutes while the scans themselves finish in seconds. `Pushing back scan job` at `count=10 limit=10` reads like a stall.

Happy to open a PR if a maintainer can say which end to fix — whether `trivy sbom` mode should be producing those reports and is failing, or whether the reuse branch should fall back to a normal scan when it yields none. I would rather ask than guess at the intended behaviour.

(Separately, and independent of this: a workload that ends up with no report is invisible in metrics either way, since every `trivy_*` series is derived from report objects that exist. #3072 and #3075 cover the two halves of that. Neither would have caught *this* bug, which falls between them.)
